markdown/docs/user-guide.md

```diff
@@ -28,370 +37,455 @@ You can start a FastBoot server on your development machine by running:
 
-Manual deployment is slow and error-prone, and even if you have a custom
-deployment process, writing an Ember CLI Deploy plugin is
-straightforward and will save you time and energy in the future.
+Manual deployment is slow and error-prone, and even if you have a custom deployment process, writing an Ember CLI Deploy plugin is straightforward and will save you time and energy in the future.
 
-As with a standard Ember build for the browser, compiled assets by
-default go in your application's `dist` directory. FastBoot adds two
-additions:
+As with a standard Ember build for the browser, compiled assets by default go in your application's `dist` directory.
+FastBoot adds two additions:
 
-1. `dist/package.json`, which contains metadata about your app for
-   consumption by the FastBoot server.
-2. `dist/fastboot/`, which contains your app's compiled JavaScript that
-   is evaluated and run by the FastBoot server.
+1. `dist/package.json`, which contains metadata about your app for consumption by the FastBoot server.
+2. `dist/fastboot/`, which contains your app's compiled JavaScript that is evaluated and run by the FastBoot server.
 
 For more information, see the [Architecture](#architecture) section.
 
-## How To...
+## The FastBoot Service
 
-### Fetch Resources Over HTTP (AJAX)
+FastBoot registers the `fastboot` service which you can inject into your application:
 
-JavaScript running in the browser relies on the `XMLHttpRequest`
-interface to retrieve resources, while Node.js offers the `http` module.
-What do we do if we want to write a single app that can fetch data from
-our API server when running in both environments?
+```javascript
+export default Ember.Route.extend({
+  fastboot: Ember.inject.service(),
+  isFastBoot: Ember.computed.reads('fastboot.isFastBoot'),
 
-One option is to use the [ember-network][ember-network] addon, which
-provides an implementation of [Fetch API][fetch-api] standard that works
-seamlessly in both environments.
+  // ... Application code
+});
+```
 
-[ember-network]: https://github.com/tomdale/ember-network
-[fetch-api]: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API
+This service contains several useful objects and methods you can use to manage an application's state when it is being FastBooted.
 
-To use `ember-network`, install it as you would any addon:
+Property         | Description
+---------------- | ---------------------------------------------------------------------------------------------------------
+`isFastBoot`     | Equals `true` if your application is running in FastBoot
+`response`       | The FastBoot server's response
+`request`        | The request sent to the FastBoot server
+`shoebox`        | A key/value store for passing data acquired server-side to the client
+`deferRendering` | A function that takes a `Promise` that you can use to defer the Ember Applications rendering in FastBoot
 
-```sh
-ember install ember-network
-```
+### Deferring Response Rendering
```

Comment:
I did have trouble with organizing this. I chose to discuss the separately since `deferRendering` is part of the FastBoot Service. 

Should I move the "why" explanation about issues of asynchronicity into the first section into this section to since the user will get to it first?